erpnext/public/js/utils.js

```diff
@@ -653,4 +653,108 @@ $(document).on('app_ready', function() {
 			});
 		});
 	}
+
+	$.each(["Purchase Receipt", "Stock Entry", "Delivery Note"], function(i, doctype) {
+		frappe.ui.form.on(doctype, {
+			refresh: function(frm) {
+				if(!frm.doc.is_return && frm.doc.status!="Closed") {
+					if (frm.doc.docstatus == 0) {
+						frm.add_custom_button(__('Make Quality Inspection'),
+						function () {
+							let data = [];
+							let inspection_type = " "
+							if (frm.doc.doctype === "Purchase Receipt") {
+								inspection_type = "Incoming";
+							}
+							else if (frm.doc.doctype === "Delivery Note") {
+								inspection_type = "Outgoing";
+							} 
+							else if (frm.doc.doctype === "Stock Entry") {
+								inspection_type = "In Process";
+							}
+							for (let row of cur_frm.doc.items) {
+								data.push({
+									"doctype": row.doctype,
+									"docname": row.name,
+									"reference_type": row.parenttype,
+									"reference_name": row.parent,
+									"item_code": row.item_code,
+									"item_name": row.item_name,
+									"inspection_type": inspection_type
+								})
+							}
+							const dialog = new frappe.ui.Dialog({
+								title: __("Make Quality Inspection"),
+								fields: [
+									{
+										label: __("Items"),
+										fieldname: "items",
+										fieldtype: "Table",
+										cannot_add_rows: true,
+										data: data,
+										in_place_edit: true,
+										get_data: () => {
+											return this.data;
+										},
+										fields: [
+											{
+												fieldtype: 'Data',
+												fieldname: "docname",
+												hidden: 1
+											},
+											{
+												label: __("Item Code"),
+												fieldtype: 'Link',
+												fieldname: "item_code",
+												options: "Item",
+												read_only: 1,
```

Comment:
The `read-only` flags are not being honoured here

![pr-edit](https://user-images.githubusercontent.com/13396535/89009128-8e230100-d329-11ea-9c9d-b5a4970c0fbe.gif)